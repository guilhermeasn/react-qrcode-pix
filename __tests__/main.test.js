import React from 'react';
import renderer from 'react-test-renderer';

import {
    PIX,
    payload
} from '../src';

import config from './config';


/* TESTANDO PAYLOAD */

test('Resultados de payload', () => {
    expect(payload(config.pixkey, config.merchant, config.city, config.cep)).toBe(config.minimalPIX);
    expect(payload(config.pixkey, config.merchant, config.city, config.cep, config.code, config.amount)).toBe(config.fullyPIX);
});


/* TESTANDO REACT */

test('Div de erro de configuracao', () => {

    const component = renderer.create(
        <PIX
            pixkey='Invalid Key'
            merchant={ config.merchant }
            city={ config.city }
        />
    ).toJSON();

    expect(component).toMatchSnapshot();
    expect(component.type).toBe('div');

});

test('QRCode com o PIX', () => {

    let payload = '';

    const component = renderer.create(
        <PIX
            pixkey={ config.pixkey }
            merchant={ config.merchant }
            city={ config.city }
            cep={ config.cep }
            code={ config.code }
            amount={ config.amount }
            onLoad={ p => payload = p }
        />
    ).toJSON();

    expect(component).toMatchSnapshot();
    expect(payload).toBe(config.fullyPIX);

});
