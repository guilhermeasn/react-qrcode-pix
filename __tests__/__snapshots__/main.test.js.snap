// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Div de erro de configuracao 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#e5d5d5",
      "border": "solid 1px",
      "borderColor": "#440000",
      "borderRadius": "10px",
      "color": "#770000",
      "display": "flex",
      "fontSize": "18px",
      "height": 256,
      "justifyContent": "center",
      "margin": "10px 0",
      "opacity": "0.75",
      "textAlign": "center",
      "width": 256,
    }
  }
>
  PIX n√£o pode ser carregado!
</div>
`;

exports[`QRCode com o PIX 1`] = `
<svg
  height={256}
  shapeRendering="crispEdges"
  viewBox="0 0 53 53"
  width={256}
>
  <path
    d="M0,0 h53v53H0z"
    fill="#FFFFFF"
  />
  <path
    d="M0 0h7v1H0zM9 0h1v1H9zM13 0h1v1H13zM16 0h1v1H16zM19 0h2v1H19zM22 0h2v1H22zM25 0h1v1H25zM27 0h1v1H27zM30 0h1v1H30zM32 0h4v1H32zM37 0h2v1H37zM40 0h3v1H40zM46,0 h7v1H46zM0 1h1v1H0zM6 1h1v1H6zM12 1h1v1H12zM15 1h1v1H15zM17 1h4v1H17zM22 1h3v1H22zM26 1h3v1H26zM30 1h1v1H30zM34 1h1v1H34zM36 1h1v1H36zM40 1h4v1H40zM46 1h1v1H46zM52,1 h1v1H52zM0 2h1v1H0zM2 2h3v1H2zM6 2h1v1H6zM14 2h4v1H14zM19 2h1v1H19zM21 2h1v1H21zM25 2h2v1H25zM28 2h1v1H28zM32 2h2v1H32zM35 2h3v1H35zM41 2h1v1H41zM43 2h1v1H43zM46 2h1v1H46zM48 2h3v1H48zM52,2 h1v1H52zM0 3h1v1H0zM2 3h3v1H2zM6 3h1v1H6zM9 3h1v1H9zM12 3h2v1H12zM16 3h3v1H16zM20 3h1v1H20zM24 3h2v1H24zM27 3h1v1H27zM29 3h1v1H29zM31 3h1v1H31zM33 3h2v1H33zM36 3h2v1H36zM39 3h1v1H39zM41 3h2v1H41zM44 3h1v1H44zM46 3h1v1H46zM48 3h3v1H48zM52,3 h1v1H52zM0 4h1v1H0zM2 4h3v1H2zM6 4h1v1H6zM9 4h3v1H9zM14 4h2v1H14zM17 4h1v1H17zM21 4h2v1H21zM24 4h8v1H24zM33 4h1v1H33zM38 4h5v1H38zM46 4h1v1H46zM48 4h3v1H48zM52,4 h1v1H52zM0 5h1v1H0zM6 5h1v1H6zM8 5h1v1H8zM10 5h4v1H10zM16 5h3v1H16zM21 5h1v1H21zM24 5h1v1H24zM28 5h1v1H28zM30 5h9v1H30zM40 5h3v1H40zM46 5h1v1H46zM52,5 h1v1H52zM0 6h7v1H0zM8 6h1v1H8zM10 6h1v1H10zM12 6h1v1H12zM14 6h1v1H14zM16 6h1v1H16zM18 6h1v1H18zM20 6h1v1H20zM22 6h1v1H22zM24 6h1v1H24zM26 6h1v1H26zM28 6h1v1H28zM30 6h1v1H30zM32 6h1v1H32zM34 6h1v1H34zM36 6h1v1H36zM38 6h1v1H38zM40 6h1v1H40zM42 6h1v1H42zM44 6h1v1H44zM46,6 h7v1H46zM10 7h2v1H10zM15 7h1v1H15zM17 7h1v1H17zM19 7h2v1H19zM24 7h1v1H24zM28 7h4v1H28zM33 7h2v1H33zM36 7h3v1H36zM43 7h1v1H43zM0 8h1v1H0zM3 8h1v1H3zM5 8h2v1H5zM8 8h3v1H8zM13 8h2v1H13zM22 8h9v1H22zM33 8h6v1H33zM40 8h3v1H40zM44 8h2v1H44zM47 8h1v1H47zM0 9h1v1H0zM2 9h1v1H2zM7 9h2v1H7zM10 9h3v1H10zM14 9h4v1H14zM20 9h1v1H20zM22 9h2v1H22zM25 9h1v1H25zM27 9h3v1H27zM33 9h1v1H33zM35 9h3v1H35zM39 9h1v1H39zM41 9h1v1H41zM45 9h1v1H45zM47 9h1v1H47zM49 9h2v1H49zM52,9 h1v1H52zM1 10h4v1H1zM6 10h3v1H6zM10 10h1v1H10zM12 10h2v1H12zM16 10h4v1H16zM21 10h1v1H21zM23 10h1v1H23zM25 10h3v1H25zM29 10h1v1H29zM31 10h7v1H31zM41 10h1v1H41zM44 10h3v1H44zM48 10h3v1H48zM1 11h1v1H1zM3 11h1v1H3zM7 11h2v1H7zM11 11h1v1H11zM13 11h1v1H13zM16 11h2v1H16zM19 11h5v1H19zM25 11h2v1H25zM29 11h2v1H29zM33 11h1v1H33zM35 11h6v1H35zM43 11h1v1H43zM47 11h2v1H47zM50 11h2v1H50zM0 12h1v1H0zM2 12h6v1H2zM9 12h1v1H9zM11 12h2v1H11zM14 12h1v1H14zM17 12h1v1H17zM19 12h7v1H19zM27 12h1v1H27zM32 12h3v1H32zM36 12h1v1H36zM39 12h2v1H39zM42 12h2v1H42zM45 12h2v1H45zM48 12h1v1H48zM52,12 h1v1H52zM0 13h2v1H0zM3 13h2v1H3zM7 13h3v1H7zM11 13h1v1H11zM14 13h2v1H14zM18 13h2v1H18zM21 13h1v1H21zM30 13h2v1H30zM33 13h2v1H33zM37 13h1v1H37zM46 13h2v1H46zM49 13h1v1H49zM52,13 h1v1H52zM0 14h1v1H0zM2 14h1v1H2zM4 14h1v1H4zM6 14h1v1H6zM9 14h1v1H9zM12 14h4v1H12zM17 14h2v1H17zM22 14h1v1H22zM28 14h4v1H28zM33 14h1v1H33zM35 14h1v1H35zM39 14h1v1H39zM44 14h1v1H44zM48 14h2v1H48zM51,14 h2v1H51zM1 15h1v1H1zM7 15h3v1H7zM12 15h4v1H12zM18 15h3v1H18zM23 15h1v1H23zM25 15h1v1H25zM27 15h1v1H27zM30 15h4v1H30zM35 15h3v1H35zM40 15h2v1H40zM44 15h2v1H44zM47 15h3v1H47zM52,15 h1v1H52zM0 16h3v1H0zM5 16h5v1H5zM11 16h1v1H11zM13 16h2v1H13zM19 16h2v1H19zM24 16h1v1H24zM32 16h2v1H32zM36 16h3v1H36zM42 16h4v1H42zM47 16h2v1H47zM51 16h1v1H51zM1 17h1v1H1zM4 17h1v1H4zM7 17h1v1H7zM9 17h2v1H9zM12 17h1v1H12zM15 17h2v1H15zM18 17h1v1H18zM22 17h1v1H22zM26 17h1v1H26zM28 17h2v1H28zM32 17h2v1H32zM35 17h1v1H35zM38 17h1v1H38zM44 17h2v1H44zM49 17h2v1H49zM52,17 h1v1H52zM1 18h1v1H1zM3 18h5v1H3zM10 18h3v1H10zM18 18h3v1H18zM22 18h1v1H22zM25 18h2v1H25zM28 18h3v1H28zM33 18h3v1H33zM37 18h2v1H37zM40 18h2v1H40zM45 18h1v1H45zM48 18h1v1H48zM50 18h1v1H50zM52,18 h1v1H52zM0 19h1v1H0zM3 19h1v1H3zM9 19h2v1H9zM12 19h1v1H12zM14 19h3v1H14zM22 19h1v1H22zM27 19h2v1H27zM31 19h2v1H31zM34 19h2v1H34zM39 19h3v1H39zM43 19h4v1H43zM48 19h1v1H48zM51 19h1v1H51zM2 20h1v1H2zM4 20h1v1H4zM6 20h1v1H6zM8 20h3v1H8zM13 20h3v1H13zM17 20h1v1H17zM19 20h1v1H19zM22 20h1v1H22zM24 20h1v1H24zM26 20h1v1H26zM29 20h2v1H29zM36 20h6v1H36zM43,20 h10v1H43zM1 21h1v1H1zM5 21h1v1H5zM7 21h1v1H7zM9 21h1v1H9zM11 21h1v1H11zM14 21h4v1H14zM19 21h1v1H19zM25 21h1v1H25zM29 21h1v1H29zM31 21h3v1H31zM36 21h2v1H36zM43 21h1v1H43zM47 21h2v1H47zM52,21 h1v1H52zM4 22h6v1H4zM11 22h2v1H11zM14 22h4v1H14zM20 22h2v1H20zM23 22h4v1H23zM29 22h1v1H29zM31 22h2v1H31zM34 22h3v1H34zM43 22h1v1H43zM45 22h6v1H45zM0 23h1v1H0zM2 23h2v1H2zM5 23h1v1H5zM8 23h2v1H8zM12 23h1v1H12zM15 23h2v1H15zM25 23h2v1H25zM30 23h4v1H30zM35 23h1v1H35zM37 23h4v1H37zM44 23h2v1H44zM48 23h1v1H48zM50,23 h3v1H50zM0 24h1v1H0zM2 24h1v1H2zM4 24h5v1H4zM11 24h1v1H11zM13 24h2v1H13zM16 24h1v1H16zM18 24h11v1H18zM30 24h3v1H30zM36 24h1v1H36zM39 24h3v1H39zM43 24h6v1H43zM50,24 h3v1H50zM1 25h1v1H1zM3 25h2v1H3zM8 25h1v1H8zM12 25h1v1H12zM18 25h1v1H18zM21 25h1v1H21zM23 25h2v1H23zM28 25h1v1H28zM31 25h2v1H31zM34 25h1v1H34zM37 25h1v1H37zM39 25h3v1H39zM43 25h2v1H43zM48 25h1v1H48zM51 25h1v1H51zM0 26h2v1H0zM4 26h1v1H4zM6 26h1v1H6zM8 26h3v1H8zM14 26h1v1H14zM17 26h1v1H17zM20 26h1v1H20zM23 26h2v1H23zM26 26h1v1H26zM28 26h2v1H28zM33 26h1v1H33zM35 26h3v1H35zM39 26h2v1H39zM44 26h1v1H44zM46 26h1v1H46zM48 26h1v1H48zM51,26 h2v1H51zM2 27h3v1H2zM8 27h2v1H8zM15 27h2v1H15zM18 27h1v1H18zM20 27h1v1H20zM23 27h2v1H23zM28 27h1v1H28zM31 27h1v1H31zM36 27h1v1H36zM38 27h1v1H38zM40 27h1v1H40zM43 27h2v1H43zM48 27h2v1H48zM51 27h1v1H51zM3 28h7v1H3zM11 28h1v1H11zM14 28h1v1H14zM18 28h1v1H18zM23 28h8v1H23zM33 28h1v1H33zM36 28h2v1H36zM41 28h1v1H41zM44 28h5v1H44zM51,28 h2v1H51zM0 29h2v1H0zM5 29h1v1H5zM7 29h5v1H7zM13 29h1v1H13zM19 29h1v1H19zM22 29h2v1H22zM27 29h1v1H27zM31 29h1v1H31zM38 29h2v1H38zM43 29h3v1H43zM47 29h1v1H47zM50 29h1v1H50zM52,29 h1v1H52zM0 30h3v1H0zM5 30h4v1H5zM11 30h2v1H11zM17 30h1v1H17zM19 30h4v1H19zM24 30h1v1H24zM26 30h2v1H26zM29 30h1v1H29zM31 30h1v1H31zM34 30h2v1H34zM41 30h1v1H41zM44 30h1v1H44zM46 30h5v1H46zM52,30 h1v1H52zM1 31h4v1H1zM7 31h1v1H7zM11 31h2v1H11zM19 31h1v1H19zM22 31h1v1H22zM25 31h1v1H25zM27 31h3v1H27zM31 31h4v1H31zM36 31h1v1H36zM39 31h1v1H39zM41 31h4v1H41zM48 31h2v1H48zM51,31 h2v1H51zM0 32h1v1H0zM4 32h1v1H4zM6 32h1v1H6zM8 32h1v1H8zM10 32h3v1H10zM17 32h1v1H17zM19 32h1v1H19zM21 32h4v1H21zM28 32h1v1H28zM30 32h1v1H30zM34 32h3v1H34zM38 32h3v1H38zM42 32h1v1H42zM44 32h1v1H44zM46 32h1v1H46zM49 32h2v1H49zM52,32 h1v1H52zM1 33h1v1H1zM4 33h1v1H4zM7 33h3v1H7zM11 33h4v1H11zM16 33h1v1H16zM22 33h1v1H22zM24 33h1v1H24zM27 33h1v1H27zM29 33h1v1H29zM31 33h3v1H31zM39 33h1v1H39zM43 33h1v1H43zM47 33h1v1H47zM52,33 h1v1H52zM0 34h1v1H0zM5 34h8v1H5zM16 34h2v1H16zM19 34h1v1H19zM24 34h3v1H24zM28 34h9v1H28zM38 34h1v1H38zM40 34h1v1H40zM43 34h2v1H43zM49 34h2v1H49zM0 35h1v1H0zM3 35h1v1H3zM5 35h1v1H5zM8 35h1v1H8zM11 35h2v1H11zM14 35h2v1H14zM17 35h2v1H17zM20 35h2v1H20zM23 35h2v1H23zM26 35h1v1H26zM31 35h1v1H31zM34 35h1v1H34zM36 35h3v1H36zM40 35h1v1H40zM44 35h1v1H44zM46 35h1v1H46zM49 35h3v1H49zM2 36h1v1H2zM6 36h1v1H6zM8 36h2v1H8zM13 36h4v1H13zM18 36h1v1H18zM20 36h1v1H20zM22 36h2v1H22zM26 36h1v1H26zM30 36h3v1H30zM35 36h3v1H35zM40 36h1v1H40zM42 36h1v1H42zM44 36h1v1H44zM46 36h4v1H46zM52,36 h1v1H52zM2 37h4v1H2zM7 37h4v1H7zM12 37h2v1H12zM16 37h1v1H16zM19 37h2v1H19zM25 37h4v1H25zM31 37h1v1H31zM36 37h2v1H36zM39 37h3v1H39zM46 37h1v1H46zM0 38h3v1H0zM4 38h1v1H4zM6 38h1v1H6zM12 38h3v1H12zM16 38h1v1H16zM18 38h1v1H18zM21 38h1v1H21zM23 38h1v1H23zM25 38h1v1H25zM27 38h1v1H27zM29 38h3v1H29zM34 38h1v1H34zM37 38h2v1H37zM40 38h1v1H40zM45 38h2v1H45zM49 38h1v1H49zM52,38 h1v1H52zM1 39h3v1H1zM7 39h1v1H7zM11 39h1v1H11zM13 39h10v1H13zM27 39h1v1H27zM29 39h1v1H29zM32 39h1v1H32zM34 39h2v1H34zM37 39h2v1H37zM40 39h2v1H40zM44 39h1v1H44zM46 39h2v1H46zM49 39h1v1H49zM52,39 h1v1H52zM0 40h4v1H0zM5 40h2v1H5zM9 40h4v1H9zM14 40h3v1H14zM21 40h2v1H21zM25 40h2v1H25zM30 40h3v1H30zM34 40h5v1H34zM41 40h3v1H41zM47 40h2v1H47zM0 41h1v1H0zM2 41h1v1H2zM4 41h1v1H4zM8 41h1v1H8zM10 41h2v1H10zM21 41h1v1H21zM23 41h1v1H23zM27 41h1v1H27zM32 41h1v1H32zM36 41h4v1H36zM44 41h2v1H44zM47 41h1v1H47zM50 41h1v1H50zM52,41 h1v1H52zM0 42h2v1H0zM3 42h4v1H3zM8 42h1v1H8zM10 42h1v1H10zM12 42h1v1H12zM14 42h3v1H14zM18 42h1v1H18zM20 42h3v1H20zM24 42h1v1H24zM26 42h4v1H26zM31 42h2v1H31zM34 42h2v1H34zM38 42h3v1H38zM46 42h1v1H46zM48 42h3v1H48zM52,42 h1v1H52zM1 43h2v1H1zM7 43h6v1H7zM15 43h1v1H15zM20 43h1v1H20zM22 43h1v1H22zM28 43h1v1H28zM33 43h3v1H33zM39 43h1v1H39zM41 43h1v1H41zM43 43h1v1H43zM49 43h1v1H49zM51,43 h2v1H51zM3 44h1v1H3zM6 44h1v1H6zM8 44h3v1H8zM13 44h1v1H13zM15 44h1v1H15zM17 44h1v1H17zM19 44h2v1H19zM22 44h7v1H22zM30 44h2v1H30zM38 44h5v1H38zM44 44h6v1H44zM8 45h1v1H8zM13 45h1v1H13zM15 45h1v1H15zM17 45h2v1H17zM20 45h1v1H20zM22 45h3v1H22zM28 45h1v1H28zM31 45h3v1H31zM35 45h3v1H35zM39 45h1v1H39zM41 45h1v1H41zM43 45h2v1H43zM48 45h2v1H48zM51,45 h2v1H51zM0 46h7v1H0zM9 46h1v1H9zM11 46h1v1H11zM13 46h5v1H13zM19 46h1v1H19zM24 46h1v1H24zM26 46h1v1H26zM28 46h2v1H28zM31 46h5v1H31zM37 46h1v1H37zM41 46h1v1H41zM44 46h1v1H44zM46 46h1v1H46zM48 46h4v1H48zM0 47h1v1H0zM6 47h1v1H6zM8 47h1v1H8zM12 47h1v1H12zM16 47h2v1H16zM20 47h1v1H20zM23 47h2v1H23zM28 47h1v1H28zM32 47h1v1H32zM36 47h3v1H36zM40 47h1v1H40zM43 47h2v1H43zM48 47h3v1H48zM0 48h1v1H0zM2 48h3v1H2zM6 48h1v1H6zM9 48h1v1H9zM12 48h2v1H12zM15 48h1v1H15zM22 48h7v1H22zM30 48h1v1H30zM32 48h2v1H32zM35 48h3v1H35zM40 48h1v1H40zM42 48h1v1H42zM44 48h6v1H44zM0 49h1v1H0zM2 49h3v1H2zM6 49h1v1H6zM8 49h4v1H8zM13 49h2v1H13zM18 49h1v1H18zM20 49h1v1H20zM22 49h4v1H22zM28 49h2v1H28zM31 49h1v1H31zM34 49h1v1H34zM37 49h1v1H37zM40 49h2v1H40zM43 49h4v1H43zM48 49h1v1H48zM51 49h1v1H51zM0 50h1v1H0zM2 50h3v1H2zM6 50h1v1H6zM13 50h5v1H13zM20 50h3v1H20zM33 50h1v1H33zM36 50h1v1H36zM38 50h3v1H38zM47 50h2v1H47zM50 50h1v1H50zM0 51h1v1H0zM6 51h1v1H6zM9 51h1v1H9zM11 51h3v1H11zM15 51h2v1H15zM24 51h1v1H24zM27 51h1v1H27zM30 51h1v1H30zM32 51h3v1H32zM36 51h2v1H36zM39 51h3v1H39zM43 51h1v1H43zM47 51h1v1H47zM51 51h1v1H51zM0 52h7v1H0zM8 52h3v1H8zM12 52h1v1H12zM15 52h6v1H15zM24 52h2v1H24zM27 52h2v1H27zM31 52h2v1H31zM36 52h2v1H36zM39 52h1v1H39zM41 52h4v1H41zM46 52h1v1H46zM48 52h2v1H48zM51 52h1v1H51z"
    fill="#000000"
  />
</svg>
`;
